### DISCUSSION TITLE
How do I run PKNCA validation?

### DISCUSSION THREAD

**Discussion by Just-another-Beth** [humanpred/pknca#303] [Q&A] [answered]
From the PKNCA Validation Vignette:

**Summary of Testing**
The following sentence is dynamically generated to summarize the testing results: Tests were not run because tests are not installed.

I don't know enough about R to run the validation tests.  When I use the other commands, it seems as if the tests were not run.  Sorry if this is a very basic question, but I feel like information

**Comment by billdenney** [comment, marked as answer]
To run the validation tests, you need to install the package with the tests, as described in the vignette:

```
install.packages(pkgs="PKNCA", INSTALL_opts="--install-tests", type="source")
```

When you do that, it will build the vignette with the tests, and it should show that all tests pass on your local system.

**Reply by Just-another-Beth** [reply]
I am very new to this, so I apologize if I just don't know what it the response would be if all the tests pass.  After I install the package, it gives me this complicated readout, and when I type "sessionInfo()" to get the results, I get more printout that doesn't seem like a test result.  Is there some kind of "build vignette" command that I'm missing.  It sounds like it should be easy, but I'm not seeing what I would expect to see.  I pasted all of my results below.

* installing *source* package 'PKNCA' ...
** package 'PKNCA' successfully unpacked and MD5 sums checked
** using staged installation
** R
** inst
** tests
** byte-compile and prepare package for lazy loading
** help
*** installing help indices
*** copying figures
** building package indices
** installing vignettes
** testing if installed package can be loaded from temporary location
** testing if installed package can be loaded from final location
** testing if installed package keeps a record of temporary installation path
* DONE (PKNCA)


sessionInfo()
Is the command, to get the information about the test, and it simply says 


> sessionInfo()
R version 4.3.3 (2024-02-29 ucrt)
Platform: x86_64-w64-mingw32/x64 (64-bit)
Running under: Windows 11 x64 (build 22621)

Matrix products: default


locale:
[1] LC_COLLATE=English_United States.utf8 
[2] LC_CTYPE=English_United States.utf8   
[3] LC_MONETARY=English_United States.utf8
[4] LC_NUMERIC=C                          
[5] LC_TIME=English_United States.utf8    

time zone: America/Chicago
tzcode source: internal

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

loaded via a namespace (and not attached):
[1] compiler_4.3.3    cli_3.6.2         tools_4.3.3       rstudioapi_0.16.0
>

**Comment by Just-another-Beth** [comment]
Sorry to be a bother.  I'm still learning to navigate in R.  My son sent me the .rnd file, and when I opened in in R studio, it ran the validation.    I can see test results in the upper right pane
<img width="389" alt="image" src="https://github.com/billdenney/pknca/assets/173188164/d47a1c62-8d38-48e4-91b3-5f32f66d6381">

Thanks for your patience

**Reply by billdenney** [reply]
I'm glad that it worked out.  If you have ideas that could make the validation test results clearer, please let me know.
